odo/backends/pandas.py

```diff
@@ -57,9 +57,20 @@ def coerce_datetimes(df):
     name            object
     dtype: object
     """
-    df2 = df.select_dtypes(include=['object']).apply(
-        partial(pd.to_datetime, errors='ignore')
-    )
+
+    def patched_to_datetime(s):
+        # NOTE: In pandas < 0.17, pd.to_datetime(' ') == datetime(...), which
+        # is not what we want.  This patched version detects empty or
+        # whitespace-only strings, and maps them to a dummy object.  This
+        # prevents datetime coercion on whitespace or empty strings.
+        o = object()
+        try:
+            s[s.str.strip() == ''] = o
+        except TypeError:
+            pass
+        return pd.to_datetime(s, errors='ignore')
+
+    df2 = df.select_dtypes(include=['object']).apply(patched_to_datetime)
```

Comment:
you ought to tests this. This i *very* inefficient. ``pd.to_datetime`` is by definition a vectorized operation ITSELF.